mongo/errors.go

```diff
@@ -184,14 +184,18 @@ func (e MongocryptdError) Unwrap() error {
 	return e.Wrapped
 }
 
+type labeledError interface {
+	error
+	// HasErrorLabel returns true if the error contains the specified label.
+	HasErrorLabel(string) bool
+}
+
 // ServerError is the interface implemented by errors returned from the server. Custom implementations of this
 // interface should not be used in production.
 type ServerError interface {
-	error
+	labeledError
```

Comment:
Because `labeledError` is an unexported type, it probably isn't a good idea to embed it in a user-facing type. I'm not sure how this would render for documentation purposes and I think it would cause more confusion than necessary. It's not ideal, but I think we may have to duplicate the `HasErrorLabel` function in both `ServerError` and `labeledError`.

Reply:
done